### PR TITLE
Feature/cc 2486

### PR DESCRIPTION
**Pull request by smousa**
Note that these changes have not been integrated yet!

* Renamed state id to HOSTID-SERVICEID-INSTANCEID to reduce the number of calls made to zookeeper and the amount of deserialization to figure out which host and service has each instance

* Minimized the size of the respective objects to only include information that was most used and most relevant to the instance.

* Using new logger schema

### CODE REVIEW

File: zzk/service/state.go [hunk 1]
```diff
@@ -0,0 +1,625 @@
...
+	s.version = version
+}
+
+// HostState2 provides information for a particular instance on host for a
+// service.
+// TODO: update name when the calls are swapped on the listeners
+type HostState2 struct {
+	DesiredState service.DesiredState
+	Scheduled    time.Time
+	version      interface{}
```
Comment (iancmcc) [inline zzk/service/state.go]: comment

File: zzk/service/state.go [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
+		InstanceID:   req.InstanceID,
+	}, nil
+}
+
+// GetServiceStates2 returns the states of a running service
+// TODO: update name when calls are swapped
+func GetServiceStates2(conn client.Connection, poolID, serviceID string) ([]State, error) {
+	// set up logging
+	logger := log.WithFields(log.Fields{
+		"ServiceID": serviceID,
```
Comment (iancmcc) [inline zzk/service/state.go]: PoolID?

File: zzk/service/state.go [hunk 3]
```diff
@@ -0,0 +1,625 @@
...
+	}
+
+	hpth := path.Join(basepth, "/hosts", hostID, "instances")
+	ch, err := conn.Children(hpth)
+	if err != nil && err != client.ErrNoNode {
+
+		// TODO: wrap the error?
+		logger.WithFields(log.Fields{
+			"Error": err,
+		}).Error("Could not look up instances on host")
```
Comment (iancmcc) [inline zzk/service/state.go]: Debug?

File: zzk/service/state.go [hunk 4]
```diff
@@ -0,0 +1,625 @@
...
+	}
+
+	spth := path.Join(basepth, "/services", serviceID)
+	ch, err := conn.Children(spth)
+	if err != nil && err != client.ErrNoNode {
+
+		// TODO: wrap the error?
+		logger.WithFields(log.Fields{
+			"Error": err,
+		}).Error("Could not look up instances on service")
```
Comment (iancmcc) [inline zzk/service/state.go]: Debug?

File: zzk/service/state.go (hunk 1 above)
Comment (smousa) [inline zzk/service/state.go]: done

**Pull request merged**